### PR TITLE
feat: add casting logic without cast-button

### PR DESCRIPTION
**Pull request by luwes**
For testing purposes there are some temporary cast buttons in the vanilla example that use the player API.
https://elements-demo-vanilla-git-fork-luwes-chromecast-mux.vercel.app/mux-player.html

- [x] ~handle loading the Google cast framework script~ decided this is up to the user to load the cast script 
- [x] upgrade Media Chrome to get the cast-button

### CODE REVIEW

File: packages/mux-player/src/template.ts
Comment (cjpillsbury) [inline packages/mux-player/src/template.ts:66]: blocking: How would you feel about CustomVideoElement just baking in using `is="castable-video"` per the discussion w/Heff on the `<media-chrome/>` side? Feels lower risk in terms of refactors and makes ChromeCast support also easier for folks using `<mux-video/>` + `<media-chrome/>` (per e.g. our TMI presentations).

File: packages/mux-video/src/CustomVideoElement.js
Comment (cjpillsbury) [inline packages/mux-video/src/CustomVideoElement.js:78]: non-blocking/followup: this is probably going to be a bit brittle, since it assumes the relevant props are on:

1. the prototype of the element and not the instance (safer for native HTMLMediaElement and related)
2. the immediate prototype of the element and not somewhere in the protochain (a bigger concern if we're anticipating e.g. mixins or other inheritance chains)

File: packages/mux-video/src/CustomVideoElement.js
Comment (cjpillsbury) [inline packages/mux-video/src/CustomVideoElement.js:244]: non-blocking: this will handle properties (with the concerns mentioned above). What about attributes? Should we e.g. check the `observedAttributes` of the element to ensure proper forwarding?

File: packages/mux-video/src/CustomVideoElement.js
Comment (cjpillsbury) [inline packages/mux-video/src/CustomVideoElement.js:263]: non-blocking: fine for now, but we should probably make this less presumptuous.

File: shared/polyfills/window.js
Comment (cjpillsbury) [inline shared/polyfills/window.js:56]: blocking: I think we're going to need to do some thorough testing to make sure we aren't breaking SSR polyfills here. If we are, this might mean we need to bump up prioritizing how we're handling polyfills on the Open Elements side of the fence.

**Comment by cjpillsbury** [comment]
blocking: It looks like this release would expose "casting logic with controls" unless I'm missing something (which is certainly possible).

**Review by cjpillsbury — CHANGES_REQUESTED** [review]
a handful of questions/callouts, with a smaller number being blockers.

File: packages/mux-video/src/CustomVideoElement.js
Comment (luwes) [reply packages/mux-video/src/CustomVideoElement.js:244]: good call, will check this out.

File: shared/polyfills/window.js
Comment (gkatsev) [reply shared/polyfills/window.js:56]: we're already polyfilling related things like HTMLElement and EventTarget above, I'm not sure that this addition will cause any issues.
Currently, the polyfills are run before the rest of mux-video and mux-player.

File: packages/mux-player/src/template.ts
Comment (luwes) [reply packages/mux-player/src/template.ts:66]: I'm fine with that but to me the current solution feels a bit easier than adding castable-video and castable-audio in each of 4 the Mux elements WC and React...

Maybe we could add Casting support later to those elements with this solution first

Comment (cjpillsbury) [reply packages/mux-player/src/template.ts:66]: ah I just meant literally updating the template of CustomVideoElement from:

```html
<slot name="media"><video part="video" crossorigin></video></slot>
```

to:
```html
<slot name="media"><video part="video" is="castable-video" crossorigin></video></slot>
```
(and importing it into the module)

Comment (cjpillsbury) [reply packages/mux-player/src/template.ts:66]: def cool with holding off as a followup if you think this is the path of lower risk

File: shared/polyfills/window.js
Comment (cjpillsbury) [reply shared/polyfills/window.js:56]: cool then marking as resolved for now. We've already run into some issues, but now that we're discussing those may have all been `document`-rlated.

**Review by cjpillsbury — APPROVED** [review]
before merging, make sure to remove the cast button (maybe comment out to make followups easier?)

File: packages/mux-video/src/CustomVideoElement.js
Comment (luwes) [inline packages/mux-video/src/CustomVideoElement.js:170]: this logic is here for a bug issued here https://github.com/muxinc/custom-video-element/issues/11

File: shared/castable-video/index.mjs
Comment (cjpillsbury) [inline shared/castable-video/index.mjs:3]: nit (non-blocking): fine for now, but we should probably export this longer term.

**Review by cjpillsbury — APPROVED** [review]
Good first pass!

**Pull request merged**
